### PR TITLE
Oauth integration

### PR DESCRIPTION
**Pull request by ansonK**
Original PR: https://github.com/ministryofjustice/defence-request-service-rota/pull/15

Adds OAuth integration for Rota app using the omniauth-dsds gem

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -24,6 +24,11 @@ gem "moj_template", "~> 0.23.0"
 gem "govuk_frontend_toolkit", "~> 2.0.1"
 gem "govuk_elements_rails", "~> 0.1.1"
 
+gem "omniauth-oauth2"
+# TODO: Go back to master when new gem version pushed
+gem "omniauth-dsds", github: "ministryofjustice/defence-request-service-omniauth-dsds", branch: "rails-app-integration"
+
+
 group :development do
```
Comment (pixeltrix) [inline Gemfile]: Remove the double line space

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,5 +1,11 @@
 class ApplicationController < ActionController::Base
+  include Omniauth::Dsds::ControllerMethods
+
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  before_action :authenticate_user!
+
+  helper_method :current_user
```
Comment (pixeltrix) [inline app/controllers/application_controller.rb]: Could these not be part of the gem module? Use ActiveSupport::Concern and a included block

File: config/routes.rb [hunk 3]
```diff
@@ -1,2 +1,7 @@
 Rails.application.routes.draw do
+  root "dashboards#show"
+
+  resource :dashboard, only: [:show]
+
```
Comment (pixeltrix) [inline config/routes.rb]: No need for a resource definition here just use get

**Comment by pixeltrix** [comment]
The tests make me :cry: but that's a bigger discussion around testing an SOA architecture than something for this PR.

Other than the minor notes I've made :+1:

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (chriscarter90) [inline app/controllers/application_controller.rb]: I agree that moving the `helper_method` defn into the gem would make sense. Wouldn't moving the `before_action` in too obscure a huge part of every controller action? I'm not sure I like the idea of having this invisible `before_action` being hidden away?

**Comment by chriscarter90** [comment]
@pixeltrix: Have actioned your comments as requested.

Will merge.

**Pull request merged**